core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.polling;
+
+import reactor.core.Disposable;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * This class provides implementation of long running operations. The poller starts polling by <b>automatically in background</b>.
+ * It has function for usual operation of a poller. For example listen to poll responses, enable/disable auto polling,
+ * manual polling, wait for polling to complete and get status of current polling.
+ * <p>
+ * Since auto polling is turned <b>on</b> by default. If some scenario requires to disable this feature.
+ * It can be done by calling setAutoPollingEnabled(false) function.
+ *
+ * <p><strong>Implementation of Long Running Operations</strong></p>
+ *
+ * @param <T>
+ */
+
+public class Poller<T> {
+
+    private String nullValueNotAllowedFormat = "Null value for %s not allowed.";
+
+    /*
+     * pollOperation is a Function that takes the previous PollResponse, and
+     * returns a new PollResponse to represent the current state
+     */
+    private Function<PollResponse<T>, Mono<PollResponse<T>>> pollOperation;
+
+    /*
+     * Various configuration options to create poller object.
+     */
+    private PollerOptions pollerOptions;
```

Comment:
Use the `final` keyword on fields that can't change after being set. I suspect a few of these can be final.

Reply:
PollerOption is gone now. I have used final in other places as needed. 
resolved